### PR TITLE
SCT-1147 add check_string fn & use in user & token classes

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/jgikbase/idmapping/util/util.py [hunk 1]
```diff
@@ -17,8 +19,39 @@ def not_none(obj: object, name: str):
...
+    if not obj:
         raise ValueError(name + ' cannot be None')
+
+# TODO EXCEP change exceptions to package specific
+
+
+_REGEX_CACHE: _Dict[str, _Pattern] = {}
+
+
+def check_string(string: str, name: str, legal_characters: str=None, max_len: int=None):
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/util/util.py]: Splitting out not_none into a `check_string` function is nice, much clearer semantics

File: src/jgikbase/idmapping/util/util.py [hunk 2]
```diff
@@ -17,8 +19,39 @@ def not_none(obj: object, name: str):
...
-        if not cast(str, obj).strip():
-            raise ValueError(name + ' cannot be None or whitespace only')
-    elif not obj:
+    if not obj:
         raise ValueError(name + ' cannot be None')
+
+# TODO EXCEP change exceptions to package specific
+
+
+_REGEX_CACHE: _Dict[str, _Pattern] = {}
```
Comment (jayrbolton) [inline src/jgikbase/idmapping/util/util.py]: I'd consider removing this, as I would bet that compiling small regexes without a cache will have zero impact on the performance of your app

File: src/jgikbase/test/idmapping/util/util_test.py [hunk 3]
```diff
@@ -1,28 +1,49 @@
...
+    check_string('mystring', 'myname')
+    check_string('foo', 'bar', max_len=3)
+    check_string('foo', 'bar', legal_characters='fo')
+    check_string('foo', 'bar', 'fo', 3)
 
 
-def test_not_none_obj_fail():
-    fail_not_none(None, 'my name', ValueError('my name cannot be None'))
+def test_check_string_fail():
+    fail_check_string(None, 'foo', None, None, ValueError('foo cannot be None'))
```
Comment (jayrbolton) [inline src/jgikbase/test/idmapping/util/util_test.py]: could add a test case where you pass a random type to check_string, such as a dict or custom object

File: src/jgikbase/idmapping/core/user.py [hunk 4]
```diff
@@ -16,8 +16,17 @@ class AuthsourceID:
...
 
+    _LEGAL_CHARS = 'a-z'
+    _MAX_LEN = 20
+
     def __init__(self, authsource: str) -> None:
-        not_none(authsource, 'authsource')
+        '''
+        Create an authorization source identifier.
+
+        :param authsource: A string identifier for the authorization source, consisting only of
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/core/user.py]: The 3 I think are likely are local, kbase, and jgi.

File: src/jgikbase/idmapping/util/util.py (hunk 2 above)
Comment (MrCreosote) [reply src/jgikbase/idmapping/util/util.py]: Probably. It also doesn't hurt, so *shrug*

File: src/jgikbase/test/idmapping/util/util_test.py (hunk 3 above)
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/util/util_test.py]: mypy will barf before the test can run

File: src/jgikbase/idmapping/core/user.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/jgikbase/idmapping/core/user.py]: Fixed authorization -> authentication. Dunno why I keep doing that

**Comment by MrCreosote** [comment]
@jayrbolton gave verbal merge approval

**Pull request merged**